### ISSUE TITLE
Issue, destreamer stops receiving packets after a few megabytes

### ISSUE THREAD

**Issue by FraH90** [snobu/destreamer#158]
Hi guys, I'm trying to download a bunch of new videos with destreamer.
Since now it has always worked... but since some days the microsoft platform seems to have a strange behaviour: first, one day he wouldn't let me log in when i launched destreamer (i tried to input the password, but it redirected me again on the form in which you need to input the password; with the same credentials however from another browser it would let me login).

Then some days later he would let me login, but I couldn't anymore download videos: it started the download, but after a few megabytes of download (1-5Mb) it stopped (no more video stream).
It's like if destreamer stopped receiving packets from the stream platform.

Tried again a lot of times, but nothing.

PS: I use the old version of destreamer, that i find more reliable in some aspects, so I'll post screens of that. But I've tried also with the new version of destreamer, and nothing: he wouldn't even download the few megabytes that the old version of destreamer does.

What's the problem? It can be that ms blocked the possibility to "register" video with destreamer?

I attach the last lines that destreamer gives me (you can wait forever, but it always get stuck there, while when was it functional he would continue receiving packets till the download was finished)

![Screenshot (246)](https://user-images.githubusercontent.com/24456169/84568365-44815580-ad7f-11ea-8d24-aff181073244.png)

Basically, it stops there, without downloading any more packets

**Comment by lukaarma** [comment]
I'll look into it, I have to run some tests but I have the suspicion that Microsoft might have changed something in their API or they are having troubles these days since also #159 is connection related

**Comment by lukaarma** [comment]
So i wasn't able to reproduce the issue, so what you can try is:

- Clone from 0 the repo
- Update ffmpeg to the latest stable build
- Check the speed and stability of your connection

Sorry if I can't tell you more but every video that i tried works without any problem

**Comment by SimoneG97** [comment]
Followed

> 
> 
> So i wasn't able to reproduce the issue, so what you can try is:
> 
>     * Clone from scratch the repo
> 
>     * Update ffmpeg to the latest stable build
> 
>     * Check the speed and stability of your connection
> 
> 
> Sorry if I can't tell you more but every video that i tried works without any problem

There's no reason to be sorry.
Just the fact that works on your end tells me that i'm an idiot and should have checked with other connections first.

Followed the instructions and got this with cabled connection:
`ffmpeg returned an error: https://euwe-1.api.microsoftstream.com/api/videos/manifest?api-version=1.0&playbackurl=https://sqvtj4ukemhuuik44nwj5wr6bh.streaming.mediaservices.windows.net/3844337b-c02b-4248-a41b-2015f079b072/1-5uzexnjuqyxqrzmo2qxxv7ngoc.ism/manifest(format=m3u8-aapl): I/O error`

Meanwhile, with mobile connections, no issues.

Which is weird since with cabled connection i can load the video on their website without any problems..
Thanks for the help, at least after trying with multiple videos i can confirm the problem should be connection related as the error log in #154 suggested

**Comment by lukaarma** [comment]
> Followed the instructions and got this with cabled connection:
`ffmpeg returned an error: https://euwe-1.api.microsoftstream.com/api/videos/manifest?api-version=1.0&playbackurl=https://sqvtj4ukemhuuik44nwj5wr6bh.streaming.mediaservices.windows.net/3844337b-c02b-4248-a41b-2015f079b072/1-5uzexnjuqyxqrzmo2qxxv7ngoc.ism/manifest(format=m3u8-aapl): I/O error`

Is there already a file where you are trying to download this video?
<br>

> Meanwhile, with mobile connections, no issues.

Have you tried again with your home network? It might have been a temporary issue
<br>
Feel free to close the issue if you don't have any other question/problem :)

**Comment by FraH90** [comment]
He can't close it I think, I've started the issue.
From my laptop I ve not resolved it.. 
It can be that Microsoft "blocked" some accounts in some way from downloading videos from ms stream? Maybe because I've downloaded too much videos?

**Comment by FraH90** [comment]
I mean, it's very strange that the script starts working doing his work and then stops abruptly without even showing an error or something

**Comment by lukaarma** [comment]
> It can be that Microsoft "blocked" some accounts in some way from downloading videos from ms stream? Maybe because I've downloaded too much videos?

I highly doubt it... If you can play the video you can download it (permission wise)

<br>

Sorry if I'm repetitive but did you try and update everything?
>From my laptop I ve not resolved it..

Because I can't understand if you did

Another suggestion I can give you is to wipe Destreamer and clone it again

**Comment by SimoneG97** [comment]
> 
> 
> > Followed the instructions and got this with cabled connection:
> > `ffmpeg returned an error: https://euwe-1.api.microsoftstream.com/api/videos/manifest?api-version=1.0&playbackurl=https://sqvtj4ukemhuuik44nwj5wr6bh.streaming.mediaservices.windows.net/3844337b-c02b-4248-a41b-2015f079b072/1-5uzexnjuqyxqrzmo2qxxv7ngoc.ism/manifest(format=m3u8-aapl): I/O error`
> 
> Is there already a file where you are trying to download this video?
> 
> > Meanwhile, with mobile connections, no issues.
> 
> Have you tried again with your home network? It might have been a temporary issue
> 
> 
> Feel free to close the issue if you don't have any other question/problem :)

No, there was no file in the folder AND i tried again with both connections, with the same results..

**Comment by lukaarma** [comment]
I've not been able to reproduce this issue in any form, even if I'm using a really bad and unstable mobile network, so I can't really tell you why/what to do :/

But I'm going to say 2 things based on you edits:

1. If ffmpeg looses connection, even for a split second, I have observed that it may take quite a long time to try again but eventually it resumes. So if your connection is not super stable/reliable it may seem like it's stuck forever (you could try and take a look at our [aria2c branch](https://github.com/snobu/destreamer/tree/aria2c_inplementation), it uses a completely different download method)

2. If the video is always the same it could be that some segments are malformed server side and when ffmpeg catches them it chokes without crashing so it stalls until somehow it skips over it (It's a wild guess, I higlhy doubt this is the reason)

**Comment by FraH90** [comment]
> I've not been able to reproduce this issue in any form, even if I'm using a really bad and unstable mobile network, so I can't really tell you why/what to do :/
> 
> But I'm going to say 2 things based on you edits:
> 
> 1. If ffmpeg looses connection, even for a split second, I have observed that it may take quite a long time to try again but eventually it resumes. So if your connection is not super stable/reliable it may seem like it's stuck forever (you could try and take a look at our [aria2c branch](https://github.com/snobu/destreamer/tree/aria2c_inplementation), it uses a completely different download method)
> 2. If the video is always the same it could be that some segments are malformed server side and when ffmpeg catches them it chokes without crashing so it stalls until somehow it skips over it (It's a wild guess, I highly doubt this is the reason)

So, I have news.
I still haven't updated ffmpeg and destreamer, i didn't have time but I'll do it now.
However, I tried to download another video (a different video) and it happens the same thing: it stops at some point, just after few megabytes.
Running it in powershell now, I was able to get the following output:
(in the beginning it stopped at 1800000)

`Opening 'crypto+https://***********************.streaming.mediaservices.windows.net/************************/1-*********************.ism/QualityLevels(414000)/Fragments(video=1800000000,format=m3u8-aapl)' for reading
[tls @ 0000016aa0c9b500] Error in the pull function.
[crypto @ 0000016aa0f35d40] Unable to open resource: https://*********************************.streaming.mediaservices.windows.net/*******************************************************/1-*****************************.ism/QualityLevels(414000)/Fragments(video=1800000000,format=m3u8-aapl)
[hls @ 0000016aa083b540] Failed to open segment 30 of playlist 0
[hls @ 0000016aa083b540] Opening 'crypto+https://***************************.streaming.mediaservices.windows.net/*******************************/1-*************************.ism/QualityLevels(414000)/Fragments(video=1860000000,format=m3u8-aapl)' for reading`

You see those errors? 
"Error in the pull function", "Unable to open resource"
"Failed to open segment 30 of playlist 0"

Basically it hangs on packet 180000 for a while, then it showed that error, and after like 5 minutes it start again downloading new packet (after the error); however the problem is that he'll do it again, he will just download a few more packet and you re again there!

I'll try later to update everything, I'll let you know if it does change anything.

**Comment by FraH90** [comment]
So.. I've updated just ffmpeg and youtube-dl (I'm using the old version of destreamer, the one that use youtube-dl, but this is not the problem cause the issue is there also with the 2.0 version of destreamer) and it still stops after a few packets. So the problem is not ffmpeg.

Now I will try updating also destreamer, first I'll do a wipe and clean install of the old version, then I'll do the same with the new version. 
Finally I will try the new branch, the aria2c branch, and I'll let you know.

**Comment by FraH90** [comment]
PS: for now this is the output I get when using destreamer with FFMPEG and youtube-dl updated:

`Opening 'crypto+https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=360000000,format=m3u8-aapl)' for reading
[tls @ 000001967a868b80] Error in the pull function.
[crypto @ 000001967c65ea40] Unable to open resource: https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=360000000,format=m3u8-aapl)
[hls @ 000001967c38f300] Failed to open segment 6 of playlist 0
[hls @ 000001967c38f300] Opening 'crypto+https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=420000000,format=m3u8-aapl)' for reading
[hls @ 000001967c38f300] Opening 'crypto+https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=480000000,format=m3u8-aapl)' for reading
[hls @ 000001967c38f300] Opening 'crypto+https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=540000000,format=m3u8-aapl)' for reading
[tcp @ 000001967a86ee80] Connection to tcp://**************************.streaming.mediaservices.windows.net:443 failed: Error number -138 occurred
[crypto @ 000001967c65ea40] Unable to open resource: https://**************************.streaming.mediaservices.windows.net/**************************/1-**************************.ism/QualityLevels(258000)/Fragments(video=540000000,format=m3u8-aapl)
[hls @ 000001967c38f300] Failed to open segment 9 of playlist 0`

**Comment by SimoneG97** [comment]
> 
> 
> I've not been able to reproduce this issue in any form, even if I'm using a really bad and unstable mobile network, so I can't really tell you why/what to do :/
> 
> But I'm going to say 2 things based on you edits:
> 
>     1. If ffmpeg looses connection, even for a split second, I have observed that it may take quite a long time to try again but eventually it resumes. So if your connection is not super stable/reliable it may seem like it's stuck forever (you could try and take a look at our [aria2c branch](https://github.com/snobu/destreamer/tree/aria2c_inplementation), it uses a completely different download method)
> 
>     2. If the video is always the same it could be that some segments are malformed server side and when ffmpeg catches them it chokes without crashing so it stalls until somehow it skips over it (It's a wild guess, I highly doubt this is the reason)

Tried to verify both points:
1. Tried the aria2c branch, still drops the error(*) with home network, no problems with mobile network (just had 1 timeout but then proceeded without problems, asking me to choose the resolution of the resulting video and starting the download).
2. If it was a server side problem, all mobile network attempts would have failed too thought, tried the same link with both solutions and all of them had this problem with home network, but worked with mobile network.

(*) Error log
`Error: timeout of 7000ms exceeded
    at createError (C:\Users\Simone\Desktop\destreamer-aria2c\node_modules\axios\lib\core\createError.js:16:15)
    at ClientRequest.handleRequestTimeout (C:\Users\Simone\Desktop\destreamer-aria2c\node_modules\axios\lib\adapters\http.js:256:16)
    at Object.onceWrapper (events.js:416:28)
    at ClientRequest.emit (events.js:322:22)
    at TLSSocket.emitRequestTimeout (_http_client.js:709:9)
    at Object.onceWrapper (events.js:416:28)
    at TLSSocket.emit (events.js:322:22)
    at TLSSocket.Socket._onTimeout (net.js:479:8)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7) {
  config: {
    url: 'videos/e22213ef-a79f-4b0f-bff3-caf836295126?api-version=1.4-private',
    method: 'get',
    headers: {
      Accept: 'application/json, text/plain, */*',
      'User-Agent': 'destreamer/2.0 (Hammer of Dawn)',
      Authorization: 'Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6IlNzWnNCTmhaY0YzUTlTNHRycFFCVEJ5TlJSSSIsImtpZCI6IlNzWnNCTmhaY0YzUTlTNHRycFFCVEJ5TlJSSSJ9.eyJhdWQiOiJodHRwczovLyoubWljcm9zb2Z0c3RyZWFtLmNvbSIsImlzcyI6Imh0dHBzOi8vc3RzLndpbmRvd3MubmV0LzlkZjA4YTdjLTMxZDctNDAyNC05YmE2LTVlZDVlZmFjMWEwMS8iLCJpYXQiOjE1OTIxNjI3ODUsIm5iZiI6MTU5MjE2Mjc4NSwiZXhwIjoxNTkyMTY2Njg1LCJhY3IiOiIxIiwiYWlvIjoiQVNRQTIvOFBBQUFBRUpvZThBemp3TUZ3c0VIWlFaOVFiTmZrZUw0NkE1Tmd3NEJOakFDVVFxVT0iLCJhbXIiOlsicHdkIl0sImFwcGlkIjoiY2Y1M2ZjZTgtZGVmNi00YWViLThkMzAtYjE1OGU3YjFjZjgzIiwiYXBwaWRhY3IiOiIyIiwiZmFtaWx5X25hbWUiOiJHQUxBTlRFIiwiZ2l2ZW5fbmFtZSI6IlNJTU9ORSIsImlwYWRkciI6IjE1MS4yNi4xNjQuMjMwIiwibmFtZSI6IlNpbW9uZSBHYWxhbnRlIiwib2lkIjoiMzNiY2Q1N2MtY2M0Ny00MWFjLWFlZjQtMmVmMWMyNDQwNTAwIiwib25wcmVtX3NpZCI6IlMtMS01LTIxLTM3OTU3MDMwNTUtMzM1MTcxMzAyNi0zMzQ1MzA0Nzc5LTE0MDk1NCIsInB1aWQiOiIxMDAzQkZGRDlBQzI0QTA5IiwicmgiOiIwLkFRd0FmSXJ3bmRjeEpFQ2JwbDdWNzZ3YUFlajhVOF8yM3V0S2pUQ3hXT2V4ejRNTUFMay4iLCJzY3AiOiJhY2Nlc3NfbWljcm9zb2Z0c3RyZWFtX3NlcnZpY2UiLCJzdWIiOiJPNi10cGlYTWZ5aFhZVEMxNzJkSUppOXRJM0VWVllQN0pUVVB6ZlhscGNZIiwidGlkIjoiOWRmMDhhN2MtMzFkNy00MDI0LTliYTYtNWVkNWVmYWMxYTAxIiwidW5pcXVlX25hbWUiOiJzaW1vbmUuZ2FsYW50ZUBzdHVkZW50LnVuaXZhcS5pdCIsInVwbiI6InNpbW9uZS5nYWxhbnRlQHN0dWRlbnQudW5pdmFxLml0IiwidXRpIjoiNDVVZmt3NXNlMGUyeUJRbnd6dEhBQSIsInZlciI6IjEuMCJ9.lJxTHgCxDLjLYhadqBsV--W6m_FPVVPFx2neSGdA-MEKWF-La9XgHRp4tQ7vyDwFaYpLx6RXA_ORXNewKT0aJ-KzwZ55-6GVGSyihiv2qh8dgo4myrwd2IfYR3n7shRkAMpCQ0QxMnE4A25drH3rVIc8JfkOqtosF9zDKjCd1pHZFdk_PxcWnWEz5urCX_XxRratj_-hW_B7XrYVJcD1dOfc6EsIAcjg56DLiJ6MupbMegWSWPxIXcmvNsg6WKTrXwDyhwZUri9tU1ftUPYwNsoT4k3TJmzeYvF7aHEWsSwI6dGT2Ok3b8lYypivm2KEFaN3-RT2rG_GccZU4g7TKA'
    },
    baseURL: 'https://euwe-1.api.microsoftstream.com/api/',
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 7000,
    adapter: [Function: httpAdapter],
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    validateStatus: [Function: validateStatus],
    'axios-retry': { retryCount: 0, lastRequestTime: 1592163086329 },
    data: undefined
  },
  code: 'ECONNABORTED',
  request: Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      bufferedRequest: null,
      lastBufferedRequest: null,
      pendingcb: 0,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: false,
      bufferedRequestCount: 0,
      corkedRequestsFree: [Object]
    },
    writable: true,
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError]
    },
    _eventsCount: 2,
    _maxListeners: undefined,
    _options: {
      protocol: 'https:',
      maxRedirects: 21,
      maxBodyLength: 10485760,
      path: '/api/videos/e22213ef-a79f-4b0f-bff3-caf836295126?api-version=1.4-private',
      method: 'GET',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: 'euwe-1.api.microsoftstream.com',
      port: null,
      nativeProtocols: [Object],
      pathname: '/api/videos/e22213ef-a79f-4b0f-bff3-caf836295126',
      search: '?api-version=1.4-private'
    },
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 0,
    _requestBodyBuffers: [],
    _onNativeResponse: [Function],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 6,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      socket: [TLSSocket],
      connection: [TLSSocket],
      _header: 'GET /api/videos/e22213ef-a79f-4b0f-bff3-caf836295126?api-version=1.4-private HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'User-Agent: destreamer/2.0 (Hammer of Dawn)\r\n' +
        'Authorization: Bearer eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6IlNzWnNCTmhaY0YzUTlTNHRycFFCVEJ5TlJSSSIsImtpZCI6IlNzWnNCTmhaY0YzUTlTNHRycFFCVEJ5TlJSSSJ9.eyJhdWQiOiJodHRwczovLyoubWljcm9zb2Z0c3RyZWFtLmNvbSIsImlzcyI6Imh0dHBzOi8vc3RzLndpbmRvd3MubmV0LzlkZjA4YTdjLTMxZDctNDAyNC05YmE2LTVlZDVlZmFjMWEwMS8iLCJpYXQiOjE1OTIxNjI3ODUsIm5iZiI6MTU5MjE2Mjc4NSwiZXhwIjoxNTkyMTY2Njg1LCJhY3IiOiIxIiwiYWlvIjoiQVNRQTIvOFBBQUFBRUpvZThBemp3TUZ3c0VIWlFaOVFiTmZrZUw0NkE1Tmd3NEJOakFDVVFxVT0iLCJhbXIiOlsicHdkIl0sImFwcGlkIjoiY2Y1M2ZjZTgtZGVmNi00YWViLThkMzAtYjE1OGU3YjFjZjgzIiwiYXBwaWRhY3IiOiIyIiwiZmFtaWx5X25hbWUiOiJHQUxBTlRFIiwiZ2l2ZW5fbmFtZSI6IlNJTU9ORSIsImlwYWRkciI6IjE1MS4yNi4xNjQuMjMwIiwibmFtZSI6IlNpbW9uZSBHYWxhbnRlIiwib2lkIjoiMzNiY2Q1N2MtY2M0Ny00MWFjLWFlZjQtMmVmMWMyNDQwNTAwIiwib25wcmVtX3NpZCI6IlMtMS01LTIxLTM3OTU3MDMwNTUtMzM1MTcxMzAyNi0zMzQ1MzA0Nzc5LTE0MDk1NCIsInB1aWQiOiIxMDAzQkZGRDlBQzI0QTA5IiwicmgiOiIwLkFRd0FmSXJ3bmRjeEpFQ2JwbDdWNzZ3YUFlajhVOF8yM3V0S2pUQ3hXT2V4ejRNTUFMay4iLCJzY3AiOiJhY2Nlc3NfbWljcm9zb2Z0c3RyZWFtX3NlcnZpY2UiLCJzdWIiOiJPNi10cGlYTWZ5aFhZVEMxNzJkSUppOXRJM0VWVllQN0pUVVB6ZlhscGNZIiwidGlkIjoiOWRmMDhhN2MtMzFkNy00MDI0LTliYTYtNWVkNWVmYWMxYTAxIiwidW5pcXVlX25hbWUiOiJzaW1vbmUuZ2FsYW50ZUBzdHVkZW50LnVuaXZhcS5pdCIsInVwbiI6InNpbW9uZS5nYWxhbnRlQHN0dWRlbnQudW5pdmFxLml0IiwidXRpIjoiNDVVZmt3NXNlMGUyeUJRbnd6dEhBQSIsInZlciI6IjEuMCJ9.lJxTHgCxDLjLYhadqBsV--W6m_FPVVPFx2neSGdA-MEKWF-La9XgHRp4tQ7vyDwFaYpLx6RXA_ORXNewKT0aJ-KzwZ55-6GVGSyihiv2qh8dgo4myrwd2IfYR3n7shRkAMpCQ0QxMnE4A25drH3rVIc8JfkOqtosF9zDKjCd1pHZFdk_PxcWnWEz5urCX_XxRratj_-hW_B7XrYVJcD1dOfc6EsIAcjg56DLiJ6MupbMegWSWPxIXcmvNsg6WKTrXwDyhwZUri9tU1ftUPYwNsoT4k3TJmzeYvF7aHEWsSwI6dGT2Ok3b8lYypivm2KEFaN3-RT2rG_GccZU4g7TKA\r\n' +
        'Host: euwe-1.api.microsoftstream.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _onPendingData: [Function: noopPendingOutput],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      insecureHTTPParser: undefined,
      path: '/api/videos/e22213ef-a79f-4b0f-bff3-caf836295126?api-version=1.4-private',
      _ended: false,
      res: null,
      aborted: true,
      timeoutCb: [Function: emitRequestTimeout],
      upgradeOrConnect: false,
      parser: [HTTPParser],
      maxHeadersCount: null,
      reusedSocket: false,
      _redirectable: [Circular],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    _currentUrl: 'https://euwe-1.api.microsoftstream.com/api/videos/e22213ef-a79f-4b0f-bff3-caf836295126?api-version=1.4-private',
    [Symbol(kCapture)]: false
  },
  response: undefined,
  isAxiosError: true,
  toJSON: [Function]
}


Unhandled error!
Timeout or fatal error, please check your downloads directory and try again`

**Comment by lukaarma** [comment]
The errors that you are seeing are not Destreamer's fault @FraH90, it's ffmpeg that can't open some crypto segments from the playlist but Destreamer doesn't touch them in any way, they are given straight to ffmpeg.
I got that same behaviour from ffmpeg when I was having trouble with my connection a couple of months ago...so other than making sure all your tools and Destreamer are up to date and your connection is stable i can't tell you more, I really can't tell anything else just from those errors :/

**Comment by lukaarma** [comment]
For your error @SimoneG97, can you check that the video you are trying to download exists? I know that it may sound a stupid question but when checking the video that Destreamer is trying to download MS Stream tells me that the content doesn't exists, not that I don't have access to it and it seems strange to me

**Comment by FraH90** [comment]
> The errors that you are seeing are not Destreamer's fault @FraH90, it's ffmpeg that can't open some crypto segments from the playlist but Destreamer doesn't touch them in any way, they are given straight to ffmpeg.
> I got that same behaviour from ffmpeg when I was having trouble with my connection a couple of months ago...so other than making sure all your tools and Destreamer are up to date and your connection is stable i can't tell you more, I really can't tell anything else just from those errors :/

You were right, seems it was an FFMPEG problem. 
I'm downloading the same videos, on the same network with the aria2c version and it doesn't stop, it seems working fine

**Comment by SimoneG97** [comment]
> 
> 
> For your error @SimoneG97, can you check that the video you are trying to download exists? I know that it may sound a stupid question but when checking the video that Destreamer is trying to download MS Stream tells me that the content doesn't exists, not that I don't have access to it and it seems strange to me

I'm sure the video exists since i can see it on the browser and download it with mobile network

**Comment by lukaarma** [comment]
The aria2c branch was created to get much better download times indeed :D

**Comment by FraH90** [comment]
> The aria2c branch was created to get much better download times indeed :D

It really does its job :D
PS don't know if you've read my second edit of the message, If it's possible, it would be nice having what I suggested in the aria2c version, the possibility of selecting a default resolution for the list of videos

**Comment by lukaarma** [comment]
@SimoneG97 maybe you have some firewall filters or your antivirus is blocking them in some way? I'm trying to make sense of such a specific error but I'm having some trouble :/

**Comment by SimoneG97** [comment]
> 
> 
> @SimoneG97 maybe you have some firewall filters or your antivirus is blocking them in some way? I'm trying to make sense of such a specific error but I'm having some trouble :/

No firewalls or antivirus (other than windows defender) are installed on this machine :\

**Comment by SimoneG97** [comment]
Ok, i think i'm lost...
Now the ffmpeg implementation just started working normally again, while the aria2c just refuses to accept any command, giving me always the same error

> `Fetching metadata...
> tmp already exists
> 
> The selected tmp directory already exists, exiting to preserve content`

**Comment by lukaarma** [comment]
@SimoneG97 
> Ok, i think i'm lost...
> Now the ffmpeg implementation just started working normally again, while the aria2c just refuses to accept any command, giving me always the same error
> 
> > `Fetching metadata...
> > tmp already exists
> > The selected tmp directory already exists, exiting to preserve content`
> 
> EDIT: Just to be clear, i didn't change URLs, nor the commands in the ffmpeg version, it's the same command that gave me problems yesterday

If the tmp folder used to store files mid download exists it will stop to preserve it, you have to delete it and all will resume normally

<br><br>
@FraH90 

> It really does its job :D
PS don't know if you've read my second edit of the message, If it's possible, it would be nice having what I suggested in the aria2c version, the possibility of selecting a default resolution for the list of videos

I know, that branch is not polished at all and still needs a lot of work, but right now I don't have the time to work on it :/
If you want just fork Destreamer and the make a new PR, help is always welcome :D

**Comment by SimoneG97** [comment]
> If the tmp folder used to store files mid download exists it will stop to preserve it, you have to delete it and all will resume normally

This solved the problem on the aria2c version, guess everything is working fine, even if i didn't change anything 😄 
Anyway, can confirm that the speed difference is **huge**, it's atleast 2-3 times faster.
Thanks for the troubleshooting!

**Comment by FraH90** [comment]
> @SimoneG97
> 
> > Ok, i think i'm lost...
> > Now the ffmpeg implementation just started working normally again, while the aria2c just refuses to accept any command, giving me always the same error
> > > `Fetching metadata... tmp already exists The selected tmp directory already exists, exiting to preserve content`
> > 
> > 
> > EDIT: Just to be clear, i didn't change URLs, nor the commands in the ffmpeg version, it's the same command that gave me problems yesterday
> 
> If the tmp folder used to store files mid download exists it will stop to preserve it, you have to delete it and all will resume normally
> 
> 
> 
> 
> @FraH90
> 
> > It really does its job :D
> > PS don't know if you've read my second edit of the message, If it's possible, it would be nice having what I suggested in the aria2c version, the possibility of selecting a default resolution for the list of videos
> 
> I know, that branch is not polished at all and still needs a lot of work, but right now I don't have the time to work on it :/
> If you want just fork Destreamer and the make a new PR, help is always welcome :D

I would love to do it, but I'm not so proficient in Javascript, neither in NodeJS (I just know the helloworld hahaha), actually I've learned the very few basics of NodeJS using destreamer, so thank you for that too ahahaha

**Comment by FraH90** [comment]
For me you can close this issue however, it's all solved using the aria2c implementation. 
(I will not do it by myself, I haven't understood if the other guy has solved it

**Comment by SimoneG97** [comment]
Yes, as i said before i'm not having any problems now so i guess you can close the issue
